### PR TITLE
Porting OMP to Parallel Utilities_10

### PR DESCRIPTION
**Pull request by ashishdarekar**
**Description**
This PR ports the OMP Parallel Loops to Parallel Utilities

Changelog
-Replaced Simple OMP parallel for loops from Kratos/linear_solver/ with IndexPartition() Parallel Utitlities

### CODE REVIEW

**Comment by ashishdarekar** [comment]
@philbucher 
Actually I am confused, How can we port simple #pragma omp parallel Block?
[like this](https://github.com/KratosMultiphysics/Kratos/blob/ec81489d42d6e2c4d0deaef3aa22ab3db7d8845f/kratos/linear_solvers/scaling_solver.h#L371)

File: kratos/linear_solvers/diagonal_preconditioner.h [hunk 1]
```diff
@@ -120,15 +120,14 @@ class DiagonalPreconditioner : public Preconditioner<TSparseSpaceType, TDenseSpa
...
 
-        #pragma omp parallel for
-        for(int i = 0 ; i < static_cast<int>(rA.size1()) ; ++i)
-        {
-            double diag_Aii = rA(i,i);
+        IndexPartition<std::size_t>(rA.size1()).for_each([&](std::size_t Index){
+            double diag_Aii = rA(Index,Index);
             if(diag_Aii != zero)
-                mDiagonal[i] = 1.00 / sqrt(fabs(diag_Aii));
+                mDiagonal[Index] = 1.00 / sqrt(fabs(diag_Aii));
```
Comment (philbucher) [inline kratos/linear_solvers/diagonal_preconditioner.h]:
```suggestion
                mDiagonal[Index] = 1.00 / std::sqrt(std::abs(diag_Aii));
```

**Review by philbucher — APPROVED** [review]
:+1:

**Comment by philbucher** [comment]
> @philbucher
> Actually I am confused, How can we port simple #pragma omp parallel Block?
> [like here](https://github.com/KratosMultiphysics/Kratos/blob/ec81489d42d6e2c4d0deaef3aa22ab3db7d8845f/kratos/linear_solvers/scaling_solver.h#L371)

we can, this would require to use the `IndexPartition` with using `ParallelUtilities::GetNumThreads` as input
=> this should go to a separate PR

**Pull request merged**